pkg/minikube/kubeconfig/config.go

```diff
@@ -17,87 +17,97 @@ limitations under the License.
 package kubeconfig
 
 import (
+	"errors"
 	"fmt"
+	"github.com/minishift/minishift/pkg/minishift/util"
+	"gopkg.in/yaml.v2"
 	"io/ioutil"
 	"os"
 	"path/filepath"
-
-	"github.com/golang/glog"
-	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
-	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api/latest"
-	"k8s.io/kubernetes/pkg/runtime"
 )
 
-// ReadConfigOrNew retrieves Kubernetes client configuration from a file.
-// If no files exists, an empty configuration is returned.
-func ReadConfigOrNew(filename string) (*api.Config, error) {
-	data, err := ioutil.ReadFile(filename)
-	if os.IsNotExist(err) {
-		return api.NewConfig(), nil
-	} else if err != nil {
-		return nil, err
-	}
+// Mock kubeconfig Data types
+type ClusterType struct {
```

Comment:
Yeah, just dealing with the raw data map[string]interface{}. Mapping into structs is one abstraction level higher. Did you not try simpleyaml or whatever it was called?